src/views/HomeView.vue

```diff
@@ -1,9 +1,158 @@
 <script setup lang="ts">
+import { ref, computed } from 'vue'
+import { useFirestore } from 'vuefire'
+import { collection, getDocs } from 'firebase/firestore'
+
+// FETCH ALL CATEGORIES AND SUBCATEGORIES
+const categories = ref<Array<{ name: string, iconURL: string, subCategories: Array<{ name: string, properties: [] }> }>>([]);
+
+const db = useFirestore()
+getDocs(collection(db, 'categories')).then((docs) => {
+  docs.forEach((doc) => {
+    console.log(doc.data());
+    categories.value.push({
+      name: doc.data().name,
+      iconURL: doc.data().icon_url,
+      subCategories: doc.data().subcategories
+    });
+  });
+});
+
+
+function numberOfRows() {
+  let total = categories.value.length;
+  let rows = 0;
+  while (total > 0) {
+    rows++;
+    total -= rows + 1; // because we start with 2 columns in the first row
+  }
+  return rows;
+}
```

Comment:
A `return Math.floor(Math.floor(Math.sqrt(1 + 8 * total) - 1) / 2);` would be slightly more optimal, but a tad less understandable, so it's up to you if you want to switch

Reply:
removed